lib/puma/cluster.rb

```diff
@@ -322,6 +343,34 @@ def setup_signals
               w.last_status[:requests_count] >= fork_requests
           end
         end
+
+      end
+      if @options[:mold_worker]
+        Signal.trap "SIGURG" do
+          mold_and_refork!
+        end
+        current_interval_index = 0
+        Signal.trap "SIGUSR1" do
+          # stop any existing mold
+          @mold&.term
+          # begin a full restart
+          phased_restart
+          # reset the mold_worker intervals
+          current_interval_index = 0
+          wakeup!
+        end
+        @events.register(:ping!) do |w|
+          # if there's a phased_restart under way don't step on its toes
+          next unless all_workers_in_phase?
+
+          next_request_interval = @options[:mold_worker][current_interval_index]
+          next unless next_request_interval && w.last_status[:requests_count] >= next_request_interval
+          # if @phase has already been incremented, we're already reforking;
+          # wait before kicking off yet another phased restart
```

Comment:
I'm don't understand the purpose of this comment. Should we be checking and comparing the current phase here?